app/cronjob/trailingTradeHelper/queue.js

```diff
@@ -5,18 +5,88 @@ const { executeTrailingTrade } = require('../index');
 const { setBullBoardQueues } = require('../../frontend/bull-board/configure');
 
 let queues = {};
+let paused = {};
+let resumed = {};
 
 const REDIS_URL = `redis://:${config.get('redis.password')}@${config.get(
   'redis.host'
 )}:${config.get('redis.port')}/${config.get('redis.db')}`;
 
+const pause = async (funcLogger, symbol) => {
+  const logger = funcLogger.child({ helper: 'queue' });
+
+  // eslint-disable-next-line no-plusplus
+  const pos = paused[symbol]++;
+  if (pos > resumed[symbol]) {
+    logger.info({ symbol }, `Queue ${symbol} pause #${pos} queued`);
+    while (pos > resumed[symbol]) {
+      // eslint-disable-next-line no-await-in-loop, no-promise-executor-return
+      await new Promise(r => setTimeout(r, 10));
+    }
+  }
+
+  await queues[symbol].pause();
+
+  logger.info({ symbol }, `Queue ${symbol} paused #${pos}`);
+};
+
+const resume = async (funcLogger, symbol) => {
+  const logger = funcLogger.child({ helper: 'queue' });
+
+  // eslint-disable-next-line no-plusplus
+  const pos = resumed[symbol]++;
+
+  if (paused[symbol] === resumed[symbol]) {
+    // eslint-disable-next-line no-multi-assign
+    paused[symbol] = resumed[symbol] = 0;
+
+    await queues[symbol].resume();
+  }
+
+  logger.info({ symbol }, `Queue ${symbol} resumed #${pos}`);
+
+  if (paused[symbol] === 0) {
+    logger.info({ symbol }, `Queue ${symbol} resumed last`);
+  }
+};
+
+const waitForJob = async (funcLogger, symbol) => {
+  const logger = funcLogger.child({ helper: 'queue' });
+
+  let active = await queues[symbol].getActiveCount();
+  if (active > 0) {
+    logger.info({ symbol }, `Queue ${symbol} wait for ${active} job(s)`);
+
+    while (active > 0) {
+      // eslint-disable-next-line no-await-in-loop, no-promise-executor-return
+      await new Promise(r => setTimeout(r, 10));
+      // eslint-disable-next-line no-await-in-loop
+      active = await queues[symbol].getActiveCount();
+    }
+  }
+
+  logger.info({ symbol }, `Queue ${symbol} inactive`);
+};
+
+const hold = async (funcLogger, symbol) => {
+  const logger = funcLogger.child({ helper: 'queue' });
+
+  if (!(symbol in queues)) {
+    logger.error({ symbol }, `No queue created for ${symbol} pause`);
+    return;
+  }
+
+  await pause(funcLogger, symbol);
+  await waitForJob(funcLogger, symbol);
+};
+
 const create = (funcLogger, symbol) => {
   const logger = funcLogger.child({ helper: 'queue' });
 
   const queue = new Queue(symbol, REDIS_URL, {
     prefix: `bull`,
     limiter: {
-      max: 1,
+      max: 100,
```

Comment:
The limit max/duration can be translated as this. Within 10 seconds, only a single job can be run.
So if there is more than a single job, the job will be put into a delayed queue. If the job is finished, it will just execute another job from the delayed queue. 

With this change, you mentioned https://github.com/chrisleekr/binance-trading-bot/issues/560#issuecomment-1368193917 that which does not help.

Can we revert it? Or is this change required?

Reply:
If you leave 1 in 10 seconds then bot will not detect any spikes in price which last less then 10 seconds. You can miss the oportunity to sell in volatile market. It was nice workaround to be sure all jobs are finished before new one is received but as I fixed the queue then there is no reason to throttle this much anymore.

We need at least 10 in 10 seconds just for price updates and then for buy and sell orders we need some reserve. If you think 100 is too much, then you can change the value, but I recommend to set not less than 20.

Reply:
> With this change, you mentioned [#560 (comment)](https://github.com/chrisleekr/binance-trading-bot/issues/560#issuecomment-1368193917) that which does not help.

There I was trying higher value to prevent queue to build up too much but I found it doesn't matter as buy/sell orders still put multiple jobs in queue. That's why it was not working in that situation as workaround. But to solve lagginess and missing spikes in market it is better to keep it higher.